### ISSUE TITLE
diff.ipynb.textconv fails with TortoiseGit on Windows

### ISSUE THREAD

**Issue by ibressler** [kynan/nbstripout#80]
When showing the diff (or patch) in TortoiseGit GUI on Windows (Anaconda3 Distribution) diff.ipynb.textconv fails with an error message:
```
Traceback (most recent call last):
  File "C:/Users/ibressle/AppData/Local/Continuum/anaconda3/lib/site-packages/nbstripout.py", line 109, in <module>
    input_stream = io.TextIOWrapper(sys.stdin.buffer, encoding='utf-8')
AttributeError: 'NoneType' object has no attribute 'buffer'
fatal: unable to read files to diff
```
To reproduce, using the Explorer context menu, right-click on a file with changes, select 'Git Commit', in the dialog, double-click the file to show the patch. In the separate patch window the error message from above appears and the presented diff contains the ipynb code with outputs respectively.

Besides, the filter for commits seems to work fine because the stripped ipynb is committed finally.
Furthermore, it works fine on the console ('Git Bash' from Explorer context menu provided by TortoiseGit).

**Comment by kynan** [comment]
Which Python version are you using? Since GUI apps don't have a stdin stream by default `sys.stdin` can be `None` in some Python versions, though [the behaviour has changed in the CPython trunk](https://github.com/python/cpython/blob/209abf746985526bce255e2fba97d3246924885d/Python/pylifecycle.c#L1787-L1798).

As long as you don't try to then read from `stdin` it should be fine not to create the `input_stream` wrapper. Are you interested to work on a pull request @ibressler ?

**Comment by kynan** [comment]
Unfortunately I don't have a Windows machine where I could test this.

**Comment by ibressler** [comment]
I had the issue with Python 3.6.5 (Anaconda3 5.2.0 64bit) on Win10. To fix this, I added pull request #81 which connects the standard streams to os.devnull if they are None initially.
Alternatively, it could also be fixed by careful handling of input_stream along the way, as you suggest (?). Setting input_stream=None if stdin is None and avoiding any code using input_stream by testing for None. Is this what you refer to?

**Comment by kynan** [comment]
I've added a comment on #81: my suggestion would have been to defer creating `input_stream` until we know it's needed and then check if `stdin` is available before creating the wrapper. I'm equally happy with your solution though!
